modules/nw-customize-ingress-error-pages.adoc

```diff
@@ -0,0 +1,104 @@
+
+Custom http error code response pages must follow the link:https://www.haproxy.com/documentation/hapee/latest/configuration/config-sections/http-errors/[HAProxy HTTP error page configuration guidelines]. Here is an example of the default {product-title} HAProxy router link:https://raw.githubusercontent.com/openshift/router/master/images/router/haproxy/conf/error-page-503.http[http 503 error code response page]. You can use the default content as a template for creating your own custom page.
+
+By default, the HAProxy router serves only a 503 error page when the application is not running or when the route is incorrect or non-existent. This default behavior is the same as the behavior on {product-title} 4.8 and earlier. If a config map for the customization of an HTTP error code response is not provided, and you are using a custom HTTP error code response page, the router serves a default 404 or 503 error code response page.
+
+[NOTE]
+====
+If you use the {product-title} default 503 error code page as a template for your customizations, the headers in the file require an editor than can use CRLF line endings.
+====
+
+.Procedure
+
+. Create a config map named `my-custom-error-code-pages` in the `openshift-config` namespace:
++
+[source,terminal]
+----
+$ oc -n openshift-config create configmap my-custom-error-code-pages \
+--from-file=error-page-503.http \
+--from-file=error-page-404.http
+----
+
+. Patch the Ingress Controller to reference the `my-custom-error-code-pages` config map by name:
++
+[source,terminal]
+----
+$ oc patch -n openshift-ingress ingresscontroller/default --patch \ '{"spec":{"httpErrorCodePages": "my-custom-error-code-pages"}}' --type=merge
+----
+
+. The Ingress Operator copies the `my-custom-error-code-pages` config map from the `openshift-config` namespace to the `openshift-ingress` namespace. A config map with name format `<your-ingresscontroller-name>-errorpages` gets created in the `openshift-ingress` namespace. Display the copy by running the following command:
++
+[source,terminal]
+----
+$ oc get cm default-errorpages -n openshift-ingress
+----
++
+.Example output
+----
+NAME                       DATA   AGE
+default-errorpages         2      25s  <1>
+----
+<1> The example config map name is `default-errorpages` because the `default` Ingress Controller custom resource (CR) was patched.
++
+The config map containing the custom error pages mounts on the router volume where the config map key is the filename that has custom http error code response.
+
+. You can verify using the one of following commands:
++
+* For 503 custom http custom error code response:
++
+[source,terminal]
+----
+$ oc -n openshift-ingress rsh <router_pod> cat /var/lib/haproxy/conf/error_code_pages/error-page-503.http
+----
++
+* For 404 custom http custom error code response:
++
+[source,terminal]
+----
+$ oc -n openshift-ingress rsh <router_pod> cat /var/lib/haproxy/conf/error_code_pages/error-page-404.http
+----
+
+.Verification
```

Comment:
nit: The preceding step begins with "You can verify..." but it precedes this block title.  My sense is that the preceding step could be the first verification step.

Reply:
I understand your sense that this should be the first verification step, but I think the Verification steps tests the code response, whereas the step you're referring to confirms the location of the mounted config map. So they are kinda separate...merp. I repurposed the paragraph that preceded this step and made it THE step. Maybe its clearer this way?